src/main/java/com/mcneilio/orcmaker/BasicEventDriver2.java

```diff
@@ -0,0 +1,144 @@
+package com.mcneilio.orcmaker;
+
+import com.mcneilio.orcmaker.columns.ColumnInitializer;
+import com.mcneilio.orcmaker.orcer.Orcer;
+import com.mcneilio.orcmaker.storage.StorageDriver;
+import com.timgroup.statsd.StatsDClient;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.exec.vector.ColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.LongColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.VectorizedRowBatch;
+import org.apache.orc.OrcFile;
+import org.apache.orc.TypeDescription;
+import org.apache.orc.Writer;
+import org.json.JSONObject;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Properties;
+import java.util.UUID;
+
+
+public class BasicEventDriver2 implements EventDriver {
+  Orcer[] orcers;
+
+  String[] colNames;
+  VectorizedRowBatch batch;
+  TypeDescription schema;
+  HashMap<String, ColumnVector> columns;
+  String eventName;
+  String eventDate;
+  String fileName;
+  String statsdEnv;
+  Configuration conf = new Configuration();
+  Writer writer = null;
+  StatsDClient statsd;
+  StorageDriver storageDriver;
+  ColumnVector[] columnVectors;
+
+
+  public BasicEventDriver2(Properties config, TypeDescription typeDescription, StorageDriver storageDriver, StatsDClient statsd) {
+    this.eventName = config.getProperty("event.name");
+    this.eventDate = config.getProperty("event.date");
+
+    this.schema = typeDescription;
+    this.storageDriver = storageDriver;
+    this.statsd = statsd;
+
+    this.batch = this.schema.createRowBatch(
+      Integer.parseInt(
+        config.getProperty("orcBatchSize", "1024")
+      )
+    );
+
+    this.columnVectors = this.batch.cols;
+    this.colNames = schema.getFieldNames().toArray(new String[0]);
+    this.orcers = new Orcer[this.batch.numCols];
+    this.columns = new HashMap<>();
+
+    ColumnInitializer.initColumns(this.columns, this.columnVectors, this.colNames, this.schema, this.orcers);
+    ColumnInitializer.nullColumns(this.columns);
+  }
+
+  @Override
+  public void addMessage(JSONObject message) {
+    long t = Instant.now().toEpochMilli();
+    int batchPosition = batch.size++;
+
+    for(int colId =0;colId<colNames.length;colId++){
+      String key=colNames[colId];
+
+      if(!message.has(key)) {
+        continue;
+      }
+      Object value = message.get(key);
+
+      if(orcers[colId]!=null){
+        orcers[colId].addObject(columnVectors[colId], batchPosition, value);
+      }
+    }
+    ((LongColumnVector) columns.get("date")).vector[batchPosition] = LocalDate.parse(eventDate).toEpochDay(); // should date be mandatory? can it be configurable?
+    columns.get("date").isNull[batchPosition] = false;
+//    statsd.count("message.count", 1, "env:"+ statsdEnv);
```

Comment:
Latest code `1.0.1` can now disable statsd!

Reply:
Nice, just saw. I added another commit to this PR with an idea to support multiple metrics clients, and the option to disable metrics altogether, without modifying code.